app/com/linkedin/drelephant/exceptions/AnalyzedLog.java

```diff
@@ -0,0 +1,172 @@
+                                if (m.find()) {
+                                        this.exception.addEventException(stringToExceptionEvent(m.group()));
+                                }
+                        }
+                }
+
+                else if(match("Finishing job [^\\s]+ attempt: [0-9]+ at [0-9]+ with status KILLED").find()){            // Killed Azkaban Job log
+                        m = match("job_[0-9]+_[0-9]+");
+                        while(m.find()){
+                                allMatches.add(m.group());
+                        }
+                        this.state = "KILLED";
+                        this.list= allMatches;
+                        //**Incomplete**
+
+                }
+
+
+                else if (match("Job failed as tasks failed").find()){                                                   // Failed MR Job log
+                        this.state = "FAILED";
+                        m = match("Task failed (task_[0-9]+_[0-9]+_[mr]_[0-9]+)");
+                        while(m.find()){
+                                allMatches.add(m.group(1));
+                        }
+                        this.list = allMatches;
+                        this.exception.setExceptionChain(matchExceptionChain(".*\\n(?:.*\\tat.+\\n)+(?:.*Caused by.+\\n(?:.*\\n)?(?:.*\\s+at.+\\n)*)*"));
+                }
+
+                else if (match("Job failed as tasks killed").find()){                                                   // Killed MR Job log
+                        this.state = "KILLED";
+                        m = match("Task killed (task_[0-9]+_[0-9]+_[mr]_[0-9]+)");
+                        while(m.find()){
+                                allMatches.add(m.group(1));
+                        }
+                        this.list = allMatches;
+                        this.exception.setExceptionChain(matchExceptionChain(".*\\n(?:.*\\tat.+\\n)+(?:.*Caused by.+\\n(?:.*\\n)?(?:.*\\s+at.+\\n)*)*"));
+
+                }
+
+                else {                                                                                                  // Failed MR Task Task log
+                        this.state="FAILED";
+                        this.exception.setExceptionChain(matchExceptionChain("Error: (.*\\n(?:.*\\tat.+\\n)+(?:.*Caused by.+\\n(?:.*\\n)?(?:.*\\s+at.+\\n)*)*)"));
+                }
+
+        }
+
+
+        private List<List<String>> matchExceptionChain (String pattern){
+                Matcher m = Pattern.compile(pattern).matcher(log);
+                List<List<String>> exceptionChain = new ArrayList<List<String>>();
+                if(m.find()){
+
+                        for (String exceptionString : stringToExceptionChain(m.group())) {
+                                exceptionChain.add(stringToExceptionEvent(exceptionString));
+                        }
+                        return exceptionChain;
+                }
+                logger.info(exceptionChain);
```

Comment:
I think printing the whole exception chain in the info log level is not a good idea. Probably put it in debug level.